h2integrate/postprocess/mapping.py

```diff
@@ -0,0 +1,852 @@
+            will raise an error if unable to resolve to one column.
+        longitude_var_name (str, optional): A string representing the column / variable name of the
+            longitude data as defined in the .csv file. Defaults to None.
+            The code will attempt to automatically detect this column if no string is provided and
+            will raise an error if unable to resolve to one column.
+        fig (plt.Figure, optional):A plt.Figure object of an existing map on which to add the
+            shipping route layer. Defaults to None.
+        ax (plt.Axes, optional): A plt.Axes object of an existing map on which to add the shipping
+            route layer. Defaults to None.
+        base_layer_gdf (
+            gpd.GeoDataFrame | list[gpd.GeoDataFrame] | tuple[gpd.GeoDataFrame, ...],
+            optional
+        ):
+            One or more GeoDataFrame objects previously used to create the base map. These are used
+            to validate CRS consistency and to calculate map bounds when adding new layers.
+            Defaults to None.
+        show_plot (bool, optional): A boolean to control whether the plot is displayed. When adding
+            multiple layers to the same figure, this should be set to False until the final layer
+            is added. Defaults to False.
+        save_plot_fpath (Path | str, optional): A string or Path object specifying where to save the
+            generated plot. If None, the plot is not saved. Defaults to None.
+        map_preferences (dict | GeospatialMapConfig, optional): A dictionary or GeospatialMapConfig
+            object defining formatting and plotting preferences for GeoPandas, matplotlib, and
+            contextily. Defaults to {}.
+
+    Returns:
+        fig (plt.Figure): The plt.Figure object for the current plot. This can be passed back into
+            the function to add additional layers.
+        ax (plt.Axes): The plt.Axes object for the current plot. This can be passed back into the
+            function to add additional layers.
+        shipping_route_gdf (gpd.GeoDataFrame): The gpd.GeoDataFrame containing the plotted shipping
+            route geometry. This can be passed back into the function to add additional layers.
+
+
+    Raises:
+        TypeError: If the provided shipping_coords_fpath is not a CSV file.
+        ValueError: If only a subset of fig, ax, and base_layer_gdf is provided.
+
+    """
+
+    # Check if map_preferences is a dictionary and load as GeospatialMapConfig.from_dict()
+    # Loads default values and checks for any erroneous / extraneous keys
+    if isinstance(map_preferences, dict):
+        map_preferences = GeospatialMapConfig.from_dict(map_preferences, strict=True)
+
+    # Load data
+    shipping_coords_fpath = find_file(shipping_coords_fpath)
+
+    # If case_results_fpath is a .csv file read in with pandas
+    if (".csv") in shipping_coords_fpath.suffix:
```

Comment:
```suggestion
    if ".csv" in shipping_coords_fpath.suffix:
```

Same comment from previous function regarding length-1 tuples.